### DISCUSSION TITLE
Is Airflow impacted by Polyfill?

### DISCUSSION THREAD

**Discussion by snowsky** [apache/airflow#40590] [General] [answered]
When I search for the keyword `polyfill`, it gave me a related package, but not sure if this is impacted by the supply chain attack.

**Comment by potiuk** [comment]
Why do you think it would be impacted ? https://sansec.io/research/polyfill-supply-chain-attack explains the package impacted, what makes you think we use it ? 

(I think it's not affected, but apparently you have an indication it is - hence my question)

**Reply by snowsky** [reply]
Found this line in package.json file, https://github.com/apache/airflow/blob/9c9706730c06c1f4eecd5e72efa5cac1a8d95d83/airflow/www/package.json#L148, not sure if it will install malicious codes.

**Reply by potiuk** [reply]
You have not mention it but I gues that vulnerability is about `polyfill` package and this one is not the one. Do you have a reason to believe it's different except that it contains "polyfill" in the name? Because you have no other reasons to suspect Airflow is vulnerable, then I would say it's a different package.

But maybe I missed something ?

**Reply by snowsky** [reply]
Great it is not related, just feel it is quiet that no one mentions or talks about this attack here, and I'm glad I've got a security credit when I asked in the other repo.

**Reply by potiuk** [reply]
> Great it is not related, just feel it is quiet that no one mentions or talks about this attack here, and I'm glad I've got a security credit when I asked in the other repo.

If you want to get security credits for reporting issues, then you should follow the security policy (click security link in the repo) and you will see that opening a public issue about dependency that has CVE is very bad idea. In order to get security credits you should privately report the issue using security@airflow.apache.org AND show how this particular public vulnerability affects project you report it with.

Most projects (including airflow) run automated tools to detect if they are using affected dependencies. Just asking public question about it without even having a clue if the project is affected is actually doing disservice to the project rather than helps with security.

If it is right, it drags attention of those who could exploit it, if it is wrong, it drags out attention of maintainers from real security issues (imagine we have 100 people asking every day 'are you affected by this and that completely unrelated issues).


So if you want to help with project's security, and gain security credits - a) follow security policy of the project. b) make an effort to actually check if the project is affected and report it when you have high confidence it is and have a Proof of Concept showing it.

Other than that such reports are just a noise and distraction.

**Reply by snowsky** [reply]
You don't get my point, security credit is not what I am looking for, but a bit surprised that Airflow community didn't respond to this severe supply chain attack, or people in this community don't understand how severe this attack is going to be?

**Reply by snowsky** [reply]
How is it hard to get a quick answer as other projects? I asked in Jupyterhub, a quick answer is enough. You don't have to write this long to explain such an entry-level concept, or even a reminder is not appreciated in this community?

**Reply by potiuk** [reply]
Why should we respond if we were unaffected apparently? You apparently have not read what I wrote - we do follow and react on security incidents. But - to be perfectly blunt - my intention here is to show an example of what happens if someone creates noise by publicly raising a question about a security issue without "doing their job" first and checking it first (and raising it properly using security policy).

We cannot afford responding to thousands people raising similar issues, so this isssue will be here showing what happens if others attempt to do the same, serving as a warning and discouraging them from doing it.

**Reply by potiuk** [reply]
I hope you wil also learn to do the homework before raising next similar issue.

**Reply by snowsky** [reply]
How did you know I did not do the homework Lol
Since you don't know the answer, we will need someone can answer this question then...

**Reply by snowsky** [reply]
> Why should we respond if we were unaffected apparently? You apparently have not read what I wrote - we do follow and react on security incidents. But - to be perfectly blunt - my intention here is to show an example of what happens if someone creates noise by publicly raising a question about a security issue without "doing their job" first and checking it first (and raising it properly using security policy).
> 
> We cannot afford responding to thousands people raising similar issues, so this isssue will be here showing what happens if others attempt to do the same, serving as a warning and discouraging them from doing it.

Are you sure 100%? Is this a certain security issue and I should report privately? let us agree to disagree.

**Reply by snowsky** [reply]
If you are on behave of Airflow or Apache Foundation, be mindful of what you have said and going to say, because it will hurt this community I believe, and will definitely benefit other competing projects and foundations.

**Reply by snowsky** [reply]
This is why I thought a discussion will help than an issue to not interrupt developers. I think you may not be technical on these stuff but triggered by a good reminder.

**Reply by snowsky** [reply]
What I'd like to confirm is that there is no hidden issue deep in the codes. It is fine that nobody can answer it now but this should be aware by the community, instead of talking about the basic rules.

**Reply by potiuk** [reply]
Well - you wrote yourself "I found pollyfil" by searching on the repo. That's not homework. That's noise. When you mention security issues it will drag attention of others. Maintainers are also very sensitive to that - when you mention security issue - you can be sure it will drag their attantion no matter if you do it as discussion or issue. If you care about security of a project do homework first and drag attention later when you are reasonably sure things are going wrong. 

If you did your homework and look it up - Airflow publlishes SBOMS (software bill of materials) - which is the standard way which is both machine and human readable which you can check if a dependency (polyfill in this case) is there: https://airflow.apache.org/docs/apache-airflow/stable/security/sbom.html  if you want to check if Airflow uses polyfill or any other dependency that has high security issue you go there and check. This is the homework and standard industry-accepted way of publishing this information.

And this is what I recommend you to do next time - hoping again that if any other person finds the discussion will do this as well.

**Reply by snowsky** [reply, marked as answer]
Hold on closing, are you 100% sure Airflow is not impacted? You haven't mentioned.

**Reply by snowsky** [reply]
Let me give you an example, https://github.com/jupyterhub/jupyterhub/security/advisories/GHSA-6rr7-c55j-m9j6. Much easier to get the answer!
